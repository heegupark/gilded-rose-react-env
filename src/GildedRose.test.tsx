import React from 'react';
import { shallow } from 'enzyme'
import GildedRose from "./GildedRose";
import ShopItemTable from "./components/ShopItemTable";
import WelcomeMessage from './components/WelcomeMessage';

describe('GildedRose', () => {
    it('should render <GildedRose />', () => {
        const component = shallow(<GildedRose />);
        expect(component.debug()).toMatchSnapshot();
        const brand = component.find('NavbarBrand');
        expect(brand.text().includes('The Gilded Rose')).toBe(true);
    });

    it('renders two <WelcomeMessage /> components', () => {
        const component = shallow(<GildedRose />);
        expect(component.find(WelcomeMessage).length).toBe(1);
    });

    it('renders two <ShopItemTable /> components', () => {
        const component = shallow(<GildedRose />);
        expect(component.find(ShopItemTable).length).toBe(2);
    });

    it('clicks Update Quality button two times and it updates the item count in tab title', () => {
        const component = shallow(<GildedRose />);
        expect(component.find({ eventKey: 'sale' }).props().title).toBe('On Sale(5)');
        expect(component.find({ eventKey: 'discount' }).props().title).toBe('Discount(0)');
        component.find('Button').simulate('click');
        component.find('Button').simulate('click');
        expect(component.find({ eventKey: 'sale' }).props().title).toBe('On Sale(4)');
        expect(component.find({ eventKey: 'discount' }).props().title).toBe('Discount(1)');
    });
});
