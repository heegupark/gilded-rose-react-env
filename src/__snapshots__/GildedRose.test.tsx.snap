// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GildedRose should render <GildedRose /> 1`] = `
"<div className=\\"App\\">
  <Container fluid={false}>
    <Row noGutters={false}>
      <Col>
        <Navbar bg=\\"light\\" expand={true} variant=\\"light\\" collapseOnSelect={false}>
          <NavbarBrand href=\\"#home\\">
            The Gilded Rose 
          </NavbarBrand>
        </Navbar>
      </Col>
    </Row>
    <Row noGutters={false}>
      <Col>
        <WelcomeMessage />
      </Col>
    </Row>
    <Row noGutters={false}>
      <Col>
        <Tabs defaultActiveKey=\\"sale\\" id=\\"uncontrolled-tab-example\\" variant=\\"tabs\\" mountOnEnter={false} unmountOnExit={false}>
          <Tab eventKey=\\"sale\\" title=\\"On Sale(5)\\">
            <Card body={false}>
              <ShopItemTable items={{...}} />
            </Card>
          </Tab>
          <Tab eventKey=\\"discount\\" title=\\"Discount(0)\\">
            <Card body={false}>
              <ShopItemTable items={{...}} />
            </Card>
          </Tab>
        </Tabs>
        <Button onClick={[Function: bound updateShowQuality]} variant=\\"primary\\" active={false} disabled={false}>
          Update Quality
        </Button>
      </Col>
    </Row>
  </Container>
</div>"
`;
